projects/components/src/table/cells/table-cell-csv-generator-management.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, Type } from '@angular/core';
+import { TableCellCsvGeneratorBase } from './table-cell-csv-generator-base';
+import { CoreTableCellCsvGeneratorType } from './types/core-table-cell-csv-generator-type';
+import { includes, isArray } from 'lodash-es';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TableCellCsvGeneratorManagementService {
+  private readonly csvGenerators: TableCellCsvGeneratorBase<unknown>[] = [];
+
+  public register(csvGenerators: TableCellCsvGeneratorBase<unknown> | TableCellCsvGeneratorBase<unknown>[]): void {
+    isArray(csvGenerators) ? csvGenerators.forEach(item => this.addGenerator(item)) : this.addGenerator(csvGenerators);
+  }
+
+  private addGenerator(csvGenerator: TableCellCsvGeneratorBase<unknown>): void {
+    if (!includes(this.csvGenerators, csvGenerator)) {
```

Comment:
nit - use your findMatchingGenerator function so we don't register two with the same type

Reply:
updated